src/renderer/tvgRender.h

```diff
@@ -258,15 +258,13 @@ struct RenderShape
 
 struct RenderEffect
 {
+    RenderMethod* rm = nullptr;
     RenderData rd = nullptr;
     RenderRegion extend = {0, 0, 0, 0};
     SceneEffect type;
     bool valid = false;
 
-    virtual ~RenderEffect()
```

Comment:
please consider the original logic. freeing the member data here seems more intuitive.